extensions/sql-database-projects/src/test/tasks/sqlDatabaseProjectTaskProvider.test.ts

```diff
@@ -0,0 +1,85 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as sinon from 'sinon';
+import * as should from 'should';
+import { SqlDatabaseProjectTaskProvider } from '../../tasks/sqlDatabaseProjectTaskProvider';
+
+describe('Sql Database Projects Task Provider', function (): void {
+	let sandbox: sinon.SinonSandbox;
+	let taskProvider: SqlDatabaseProjectTaskProvider;
+
+	// Define a mock workspace folder for testing
+	const workspaceFolder: vscode.WorkspaceFolder = {
+		uri: vscode.Uri.file('/SqlProjFolder'),
+		name: 'SqlProjFolder',
+		index: 0
+	};
+
+	// Define a mock .sqlproj file URI for testing
+	const sqlProjUri = vscode.Uri.file('/SqlProjFolder/MyProject/MyProject.sqlproj');
+
+	// Helper to stub VS Code workspace APIs for consistent test environment
+	function stubWorkspaceAndFiles() {
+		sandbox.stub(vscode.workspace, 'workspaceFolders').value([workspaceFolder]);
+		sandbox.stub(vscode.workspace, 'findFiles').resolves([sqlProjUri]);
+	}
+
+	beforeEach(() => {
+		// Create a new Sinon sandbox before each test
+		sandbox = sinon.createSandbox();
+		// Instantiate the task provider with the mock workspace folder
+		taskProvider = new SqlDatabaseProjectTaskProvider([workspaceFolder]);
+	});
+
+	afterEach(() => {
+		// Restore the Sinon sandbox and any stubs after each test
+		sandbox.restore();
+	});
+
+	it('Should create build and buildWithCodeAnalysis tasks for .sqlproj files with correct properties', async function (): Promise<void> {
```

Comment:
Let's create 2 more test methods:
- 1 that tests the negative case (when there's no .sqlproj)
- 1 that tests when 2 .sqlproj files are in the workspace